daffodil-lib/src/main/scala/org/apache/daffodil/lib/exceptions/Assert.scala

```diff
@@ -128,7 +128,7 @@ object Assert extends Assert {
   //
 
   def usageError(message: String = "Usage error."): Nothing = {
-    abort(message)
+    toss(new IllegalStateException(message))
```

Comment:
I don't think of a usage error as an illegal state exception. If you want to use a precise Java exception class for this I think there is an IllegalArgumentException.

I *do* think of an invariant failure as an Illegal state exception. 

But we have quite a bit of code depending on the fact that we always throw only one of our own exceptions so that they can be handled uniformly (or explicitly NOT handled when somebody wants to be sure they are not being captured.)

So I think the right thing is to use a Java IllegalArgumentException as the cause of a Daffodil-specific exception like Abort. 

Let's improve the scaladoc on usage-related assert methods. They're about checking args. 

They're also primarily intended for *us* daffodil developers using our own internal methods.